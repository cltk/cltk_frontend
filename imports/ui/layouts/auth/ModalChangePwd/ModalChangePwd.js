import React from 'react';
import { Meteor } from 'meteor/meteor';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

// lib
import muiTheme from '/imports/lib/muiTheme';

const ModalChangePwd = React.createClass({

	propTypes: {
		lowered: React.PropTypes.bool,
		closeModal: React.PropTypes.func,
	},

	childContextTypes: {
		muiTheme: React.PropTypes.object.isRequired,
	},

	getChildContext() {
		return { muiTheme: getMuiTheme(muiTheme) };
	},

	componentWillMount() {
		document.addEventListener('keydown', this._handleKeyDown);
	},

	componentWillUnmount() {
		document.removeEventListener('keydown', this._handleKeyDown);
	},

	_handleKeyDown(event) {

		const { closeModal } = this.props;

		if (event.keyCode === 'ESCAPE_KEY') closeModal();
	},

	render() {
		const lowered = this.props.lowered;

		return (
			<div>
				{Meteor.userId() ?
					<div
						className={`cltk-modal-login
						cltk-modal cltk-login-signup ${((lowered) ? ' lowered' : '')}`}
					>
						<div
							className="close-modal paper-shadow"
							onClick={this.props.closeModal}
						>
							<i className="mdi mdi-close" />
						</div>
						<div className="modal-inner">
							<BlazeToReact blazeTemplate="atForm" state="changePwd" />
						</div>
					</div>
					: ''
				}
			</div>
		);
	},
});

export default ModalChangePwd;
