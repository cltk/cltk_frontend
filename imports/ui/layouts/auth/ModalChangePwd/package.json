{
  "name": "ModalChangePwd",
  "version": "0.0.1",
  "private": true,
  "main": "./ModalChangePwd.js"
}
