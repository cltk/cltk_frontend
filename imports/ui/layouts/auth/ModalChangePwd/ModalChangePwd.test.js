import React from 'react';
import renderer from 'react-test-renderer';

// component:
import ModalChangePwd from './ModalChangePwd';

describe('ModalChangePwd', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<ModalChangePwd />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
