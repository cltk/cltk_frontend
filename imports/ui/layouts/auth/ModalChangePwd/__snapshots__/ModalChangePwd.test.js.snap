// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalChangePwd renders correctly 1`] = `
<div>
  
</div>
`;
